qiskit_nature/properties/grouped_property.py

```diff
@@ -70,6 +70,24 @@ def add_property(self, prop: Optional[T_co]) -> None:
                 name = prop.__class__.__name__
             self._properties[name] = prop
 
+    def remove_property(self, prop: Union[str, Type[Property], Property]) -> Optional[T_co]:
+        """Removes a property from the group.
+
+        Args:
+            prop: the name or type of the property to remove from the group.
+
+        Returns:
+            The removed property (or None).
+        """
+        name: str
+        if isinstance(prop, str):
+            name = prop
+        elif isinstance(prop, Property):
+            name = prop.name
```

Comment:
No, it is not needed. I put it there because e.g. for the `DipoleMoment` property `prop.name` and `DipoleMoment.__name__` are different. I will remove these two lines and directly pass the name of a property to `remove_property`,
```python
transformed_property.remove_property(internal_property.name)
```

Reply:
Ah that's a good point yes!